### DISCUSSION TITLE
Reproducibility Package Plan

### DISCUSSION THREAD

**Discussion by Wes-Mac** [ecn310/course-project-zipcentercrime#18] [General]
This is where we will brainstorm for our reproducibility package

**Comment by Gersh305** [comment]
Requirements from Professor Buzard's slides [here](https://worldbank.github.io/dime-data-handbook/publication.html#box-7.1-summary-publishing-reproducible-research-outputs)

1. Raw data should:
  a. not be modified in any way;
  b. contain all observations originally collected in a widely accessible format, with a data codebook describing all variables;
  c. have original documentation about the collection of the data, such as API script or step-by-step instructions, in EXCRUCIATING detail;
2. Research reproducibility package should contain
  a. all code required to derive analysis data from downloaded data;
  b. all code required to reproduce research outputs from analysis data;
  c. a README with documentation on the use and structure of the code;
    i. should explain any intermediate datasets
    ii. should give location and how to access all files, including figures
3. The research report should
  a. be written and maintained as a dynamic LaTeX file
  b. be written in Overleaf
    i. .tex file can be pushed to Github from Overleaf
    ii. You’ll have to download the PDF from Overleaf and upload to Github once the report is final;
  c. [in a Data Appendix] contain instructions about where to find / how to access your reproducibility package (all your documentation, data and code);
  d. be recompiled with all final figures, tables, and other code outputs before submission.

**Comment by Wes-Mac** [comment]
- Do files for splitting raw data by year, and do file for editing data from ArcGIS
- File for ArcGIS steps

**Comment by regaudre** [comment]
### To-do
- [ ]  Create Codebook for SAMHSA raw data - Sophia, by 11/22
- [x] Create Presentation for Thursday, 11/21
- [ ] organize and create step by step instructions to reproduce analysis
- [ ] weed out any extra do files or data sets and make sure it reflects in [Master Documentation file](https://github.com/ecn310/course-project-zipcentercrime/blob/main/Master%20Documentation%20File.md) - Rachel, by 11/21
- [ ] add `log` to every do file - Weston, by 12/3
- [ ] improve instructions and documentation in do files - Weston, by 12/5
- [ ] create README - Weston , by 12/3
- [ ] Have all files (data, do, and logs) stored in a central folder within the package with the readme - Weston, by 12/5

**Reply by kbuzard** [reply]
@ecn310/zipcentercrime 
- "organize and create step by step instructions to reproduce analysis" is a central task here and it doesn't have anyone assigned to do it or a due date.
  - How is this different from "create README"? You'll need a high-level README on the main repo page--is this what you're talking about, and the above will be the README on the page for the replication package folder?
- There's nothing here about your figures and other research output. Where will those live?
- If you're keeping this separate from your main project plan (it's fine either way), then you should have something here that repeats the key targets there (inside testing, outside testing, revision, etc). 

> add log to every do file

Remember that you need this for all the current do files for tomorrow's documentation check
